internal/object/object.go

```diff
@@ -230,26 +217,29 @@ func (o *Object) EvalExists(key string) bool {
 		return true
 	}
 
-	for _, m := range o.Metadata {
-		_, ok = m.ExtraTags[key]
-		if ok {
-			return true
-		}
+	_, ok = o.ExtraTags[key]
+	if ok {
+		return true
 	}
 
 	return false
 }
 
 // TODO: the return value of this function must be stable like forever
-func ID(tags map[string]string) types.Binary {
+func (o *Object) ID() types.Binary {
```

Comment:
Why did you make this a method on `*Object`? The ID is supposed to be a deterministic value computed from a limited set of inputs, so I think it's a good idea to keep those as explicit parameters.